lxd/patches.go

```diff
@@ -141,3 +145,816 @@ func patchNetworkPermissions(name string, d *Daemon) error {
+		//  Simply rename the container when they are
+		// directories.
+		oldContainerMntPoint := shared.VarPath("containers", ct)
+		newContainerMntPoint := getContainerMountPoint(defaultPoolName, ct)
+		err = os.Rename(oldContainerMntPoint, newContainerMntPoint)
+		if err != nil {
+			return err
+		}
+
+		doesntMatter := false
+		err = createContainerMountpoint(newContainerMntPoint, oldContainerMntPoint, doesntMatter)
+		if err != nil {
+			return err
+		}
+
+		//  Check if we need to account for snapshots
+		// for this container.
+		oldSnapshotMntPoint := shared.VarPath("snapshots", ct)
+		if !shared.PathExists(oldSnapshotMntPoint) {
+			continue
+		}
+
+		//  If the snapshots directory for that
+		// container is empty, remove it.
+		isEmpty, err := shared.PathIsEmpty(oldSnapshotMntPoint)
+		if isEmpty {
+			os.Remove(oldSnapshotMntPoint)
+			continue
+		}
+
+		//  Create the new path where snapshots will
+		// be located on the new storage api.
+		snapshotsMntPoint := shared.VarPath("storage-pools", defaultPoolName, "snapshots")
+		err = os.MkdirAll(snapshotsMntPoint, 0711)
+		if err != nil {
+			return err
+		}
+
+		//  Now simply rename the snapshots directory
+		// as well.
+		newSnapshotMntPoint := getSnapshotMountPoint(defaultPoolName, ct)
+		err = os.Rename(oldSnapshotMntPoint, newSnapshotMntPoint)
+		if err != nil {
+			return err
+		}
+
+		//  Create a symlink for this container.
+		// snapshots.
+		err = createSnapshotMountpoint(newSnapshotMntPoint, newSnapshotMntPoint, oldSnapshotMntPoint)
+		if err != nil {
+			return err
+		}
+	}
+
+	//  Insert storage volumes for snapshots into the
+	// database. Note that snapshots have already been moved and symlinked
+	// above. So no need to do any work here.
+	for _, cs := range cSnapshots {
+		_, err := dbStoragePoolVolumeCreate(d.db, cs, storagePoolVolumeTypeContainer, poolID, volumeConfig)
+		if err != nil {
+			shared.LogWarnf("Could not insert a storage volume for snapshot \"%s\".", cs)
+			continue
+		}
+	}
+
+	//  Insert storage volumes for images into the
+	// database. Images don't move. The tarballs remain in their original
+	// location.
+	images := append(imgPublic, imgPrivate...)
+	for _, img := range images {
+		_, err := dbStoragePoolVolumeCreate(d.db, img, storagePoolVolumeTypeImage, poolID, volumeConfig)
+		if err != nil {
+			shared.LogWarnf("Could not insert a storage volume for image \"%s\".", img)
+			continue
+		}
+	}
+
+	return nil
+}
+
+func upgradeFromStorageTypeLvm(name string, d *Daemon, defaultPoolName string, defaultStorageTypeName string, cRegular []string, cSnapshots []string, imgPublic []string, imgPrivate []string) error {
+	poolConfig := map[string]string{}
+	poolConfig["source"] = defaultPoolName
+	poolConfig["volume.lvm.thinpool_name"] = daemonConfig["storage.lvm_thinpool_name"].Get()
+	poolConfig["volume.block.filesystem"] = daemonConfig["storage.lvm_fstype"].Get()
+	poolConfig["volume.block.mount_options"] = daemonConfig["storage.lvm_mount_options"].Get()
```

Comment:
I'm not seeing any code that clears those after migration, did I miss something or are we ending up with global config keys that are set and can't ever be cleared since they're now marked deprecated?